Mathlib/Topology/Algebra/InfiniteSum/Basic.lean

```diff
@@ -25,526 +25,605 @@ open scoped BigOperators Topology
+    (hs : (s : Set ι).Pairwise (Disjoint on t)) (hf : ∀ i ∈ s, HasProd (f ∘ (↑) : t i → α) (a i)) :
+    HasProd (f ∘ (↑) : (⋃ i ∈ s, t i) → α) (∏ i in s, a i) := by
+  simp_rw [hasProd_subtype_iff_mulIndicator] at *
+  rw [Finset.mulIndicator_biUnion _ _ hs]
+  exact hasProd_prod hf
 #align has_sum_sum_disjoint hasSum_sum_disjoint
 
-theorem HasSum.add_isCompl {s t : Set β} (hs : IsCompl s t) (ha : HasSum (f ∘ (↑) : s → α) a)
-    (hb : HasSum (f ∘ (↑) : t → α) b) : HasSum f (a + b) := by
+@[to_additive]
+theorem HasProd.mul_isCompl {s t : Set β} (hs : IsCompl s t) (ha : HasProd (f ∘ (↑) : s → α) a)
+    (hb : HasProd (f ∘ (↑) : t → α) b) : HasProd f (a * b) := by
   simpa [← hs.compl_eq] using
-    (hasSum_subtype_iff_indicator.1 ha).add (hasSum_subtype_iff_indicator.1 hb)
+    (hasProd_subtype_iff_mulIndicator.1 ha).mul (hasProd_subtype_iff_mulIndicator.1 hb)
 #align has_sum.add_is_compl HasSum.add_isCompl
 
-theorem HasSum.add_compl {s : Set β} (ha : HasSum (f ∘ (↑) : s → α) a)
-    (hb : HasSum (f ∘ (↑) : (sᶜ : Set β) → α) b) : HasSum f (a + b) :=
-  ha.add_isCompl isCompl_compl hb
+@[to_additive]
+theorem HasProd.mul_compl {s : Set β} (ha : HasProd (f ∘ (↑) : s → α) a)
+    (hb : HasProd (f ∘ (↑) : (sᶜ : Set β) → α) b) : HasProd f (a * b) :=
+  ha.mul_isCompl isCompl_compl hb
 #align has_sum.add_compl HasSum.add_compl
 
-theorem Summable.add_compl {s : Set β} (hs : Summable (f ∘ (↑) : s → α))
-    (hsc : Summable (f ∘ (↑) : (sᶜ : Set β) → α)) : Summable f :=
-  (hs.hasSum.add_compl hsc.hasSum).summable
+@[to_additive]
+theorem Multipliable.mul_compl {s : Set β} (hs : Multipliable (f ∘ (↑) : s → α))
+    (hsc : Multipliable (f ∘ (↑) : (sᶜ : Set β) → α)) : Multipliable f :=
+  (hs.hasProd.mul_compl hsc.hasProd).multipliable
 #align summable.add_compl Summable.add_compl
 
-theorem HasSum.compl_add {s : Set β} (ha : HasSum (f ∘ (↑) : (sᶜ : Set β) → α) a)
-    (hb : HasSum (f ∘ (↑) : s → α) b) : HasSum f (a + b) :=
-  ha.add_isCompl isCompl_compl.symm hb
+@[to_additive]
+theorem HasProd.compl_mul {s : Set β} (ha : HasProd (f ∘ (↑) : (sᶜ : Set β) → α) a)
+    (hb : HasProd (f ∘ (↑) : s → α) b) : HasProd f (a * b) :=
+  ha.mul_isCompl isCompl_compl.symm hb
 #align has_sum.compl_add HasSum.compl_add
 
-theorem Summable.compl_add {s : Set β} (hs : Summable (f ∘ (↑) : (sᶜ : Set β) → α))
-    (hsc : Summable (f ∘ (↑) : s → α)) : Summable f :=
-  (hs.hasSum.compl_add hsc.hasSum).summable
+@[to_additive]
+theorem Multipliable.compl_add {s : Set β} (hs : Multipliable (f ∘ (↑) : (sᶜ : Set β) → α))
+    (hsc : Multipliable (f ∘ (↑) : s → α)) : Multipliable f :=
+  (hs.hasProd.compl_mul hsc.hasProd).multipliable
 #align summable.compl_add Summable.compl_add
 
-/-- Version of `HasSum.update` for `AddCommMonoid` rather than `AddCommGroup`.
+@[to_additive "Version of `HasSum.update` for `AddCommMonoid` rather than `AddCommGroup`.
 Rather than showing that `f.update` has a specific sum in terms of `HasSum`,
```

Comment:
Why is there no multiplicative docstring here?

Reply:
The reason is silly. It is because I multiplicativized this file before multiplicativizing on `Topology.Algebra.InfiniteSum.Group`, so the lemma `HasProd.update`, which I needed to reference in the multiplicative docstring, didn't exist yet. Thank you for catching this.